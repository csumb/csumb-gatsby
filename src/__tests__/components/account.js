import React from 'react'
import renderer from 'react-test-renderer'
import {
  AccountSidebar,
  AccountTitle,
  AccountData,
  AccountGroup,
} from 'components/pages/account'

const profile = {
  firstName: 'Joe',
  lastName: 'Otter',
}

const staffUser = {
  _isEmployee: true,
  profile: profile,
}

const applicantUser = {
  _isApplicant: true,
  profile: profile,
}

describe('AccountSidebar', () => {
  it('renders the sidebar for Staff', () => {
    const tree = renderer.create(<AccountSidebar user={staffUser} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders the sidebar for Applicants', () => {
    const tree = renderer
      .create(<AccountSidebar user={applicantUser} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('AccountTitle', () => {
  it('renders account title correctly', () => {
    const tree = renderer
      .create(<AccountTitle>The title</AccountTitle>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('AccountData', () => {
  it('renders account data correctly', () => {
    const tree = renderer.create(<AccountData>Some data</AccountData>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('AccountGroup', () => {
  it('renders account data groups correctly', () => {
    const tree = renderer
      .create(<AccountGroup legend="This is a legend">Some data</AccountGroup>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
