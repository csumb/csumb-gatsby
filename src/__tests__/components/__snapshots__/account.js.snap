// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountData renders account data correctly 1`] = `
<p
  className="css-h9i116"
>
  Some data
</p>
`;

exports[`AccountGroup renders account data groups correctly 1`] = `
<div
  className="css-ade9ts"
>
  <h3
    className="css-1kiihh0"
  >
    This is a legend
  </h3>
  Some data
</div>
`;

exports[`AccountSidebar renders the sidebar for Applicants 1`] = `
<ul
  className="css-1rp3yp7"
>
  <a
    className="css-rvxwuc"
    href="/account"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Your account
  </a>
  <a
    className="css-rvxwuc"
    href="/account/applicant-status"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Your application status
  </a>
  <a
    className="css-rvxwuc"
    href="/account/card"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Otter Card
  </a>
  <a
    className="css-rvxwuc"
    href="/account/print"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Printer paper
  </a>
</ul>
`;

exports[`AccountSidebar renders the sidebar for Staff 1`] = `
<ul
  className="css-1rp3yp7"
>
  <a
    className="css-rvxwuc"
    href="/account"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Your account
  </a>
  <a
    className="css-rvxwuc"
    href="/account/profile"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Your public profile
  </a>
  <a
    className="css-rvxwuc"
    href="/account/card"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Otter Card
  </a>
  <a
    className="css-rvxwuc"
    href="/account/print"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Printer paper
  </a>
</ul>
`;

exports[`AccountTitle renders account title correctly 1`] = `
<h2
  className="css-jk6p7n"
>
  The title
</h2>
`;
